maths/rsa.py

```diff
@@ -23,66 +23,72 @@
 
 import random
 
-from rabin_miller import * # is_prime
-from extended_gcd import * # extended_gcd
 
-"""
-generate a prime with k bits
-"""
-def genprime(k):
-    random.seed()
-    while True:
-        n = random.randrange(int(2 ** (k - 1)),int(2 ** k))
-        if is_prime(n,128):
-            return n
-
-
-"""
-calculate the inverse of a mod m
-that is, find b such that (a * b) % m == 1
-"""
 def modinv(a, m):
-        b = 1
-        while ((a*b) % m != 1):
-            b += 1
-        return b
+    """calculate the inverse of a mod m
+    that is, find b such that (a * b) % m == 1"""
+    b = 1
+    while not (a * b) % m == 1:
+        b += 1
+    return b
+
+
+def generate_key(k, seed=None):
+    """
+    the RSA key generating algorithm
+    k is the number of bits in n
+    """
+
+    def gen_prime(k, seed=None):
+        """generate a prime with k bits"""
+
+        def is_prime(num):
+            if num == 2:
+                return True
+            for i in range(2, int(num ** 0.5) + 1):
+                if num % i == 0:
+                    return False
+            return True
+
+        random.seed(seed)
```

Comment:
Instead of ```random.seed(seed)``` you can simple write ```random.seed()``` . Then you can remove the unnecessary argumente of the function.

Reply:
We have to provide a _seed_, because we need to create the similar random numbers for testing purposes. If we just call `random.seed()` without argument, it takes the system time as _seed_, which obviously varies, therefore I wrote `random.seed(seed)`.